test/e2e/client.go

```diff
@@ -127,6 +139,42 @@ func (c *E2ETestClient) Get(ctx context.Context, key types.NamespacedName, obj c
 	return c.Client.Get(ctx, key, obj)
 }
 
+func (c *E2ETestClient) Execute(pod corev1.Pod, containerName, command string) (string, error) {
+	req := c.CoreV1Client.RESTClient().
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: containerName,
+			Command:   []string{"/bin/sh", "-c", command},
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+
+	buf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+	exec, err := remotecommand.NewSPDYExecutor(c.restConfig, "POST", req.URL())
+	if err != nil {
+		panic(err)
+	}
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdout: buf,
+		Stderr: errBuf,
+	})
+	if err != nil {
+		return "", fmt.Errorf(`failed executing command "%s" on %v/%v: %s ("%s")`, command, pod.Namespace, pod.Name, err, errBuf.String())
```

Comment:
Q: Wondering if we need to return the `errBuf` as well, `err` should contain it right?

Reply:
No strong opinion here, but if you think `errBuf` doesn't have any explicit useful info, I am slightly in favor of just returning the `err`.